### PR TITLE
return raw effects for dev inspect

### PR DESCRIPTION
**Pull request by emmazzz**
## Description 

Similar to #15790, raw txn effects is also needed to implement dry run for graphql.

## Test Plan 

existing tests.

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 1]
```diff
@@ -1050,8 +1050,8 @@ pub struct DevInspectArgs {
     pub gas_objects: Option<Vec<ObjectRef>>,
     /// Whether to skip transaction checks for the transaction.
     pub skip_checks: Option<bool>,
-    /// Whether to return the raw transaction data.
-    pub show_raw_txn_data: Option<bool>,
+    /// Whether to return the raw transaction data and effects.
+    pub show_raw_txn_data_and_effects: Option<bool>,
```
Comment (mwtian) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:1054]: Is there any compatibility issue with this change?

**Review by mwtian — APPROVED** [review]

File: crates/sui-json-rpc-types/src/sui_transaction.rs (hunk 1 above)
Comment (emmazzz) [reply crates/sui-json-rpc-types/src/sui_transaction.rs:1054]: Should be backwards compatible since it's an `Option`

Comment (amnn) [reply crates/sui-json-rpc-types/src/sui_transaction.rs:1054]: I guess the question here is whether there's an issue for people who were using this flag before just to get the raw tx data and now they get both data and effects, but I'm hoping/guessing that you added this flag recently and so we (GraphQL) are the only customer?

**Review by amnn — APPROVED** [review]
LGTM as well, thanks for spotting and fixing @emmazzz !

File: crates/sui-json-rpc-types/src/sui_transaction.rs (hunk 1 above)
Comment (emmazzz) [reply crates/sui-json-rpc-types/src/sui_transaction.rs:1054]: Correct. This is only two days old.

**Pull request merged**